### PR TITLE
Deprecate ParmedConverter taking Timestep objects

### PR DESCRIPTION
**Pull request by lilyminium**
~Fixes #~
Relates to #3031 

Changes made in this Pull Request:
 - Fixed ParmedConverter reading Timestep objects, at least the AttributeErrors
 - Deprecated ParmedConverter reading Timestep objects

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/coordinates/ParmEd.py [hunk 1]
```diff
@@ -211,9 +215,21 @@ def convert(self, obj):
         atom_kwargs = []
         for atom, name, resname, xyz, vel in zip(ag_or_ts, names, resnames, 
                                                  positions, velocities):
-            akwargs = {'name': name}
-            chain_seg = {'segid': atom.segid}
-            for attrname in ('mass', 'charge', 'type',
+            try:
```
Comment (IAlibay) [inline package/MDAnalysis/coordinates/ParmEd.py:218]: I'll be honest, I'm not sure why this is being added here, was `ts` just not working to begin with? If so, do we even need to deprecate it (there's 0 end-user impact except from us removing a dead code path right?)

Comment (lilyminium) [reply package/MDAnalysis/coordinates/ParmEd.py:218]: But I want it to work (or at least not die) in 1.1 :-(

Comment (IAlibay) [reply package/MDAnalysis/coordinates/ParmEd.py:218]: 🙃 we're going to need to test that the outcome makes sense then.. or you can burn it with fire.

Comment (lilyminium) [reply package/MDAnalysis/coordinates/ParmEd.py:215]: Too much work for something that's a little silly (making a template molecule with atoms called X, masses of 0, resids of 1...) and will be removed next release anyway.

File: package/MDAnalysis/coordinates/ParmEd.py [hunk 2]
```diff
@@ -211,9 +212,21 @@ def convert(self, obj):
...
```
Comment (IAlibay) [inline package/MDAnalysis/coordinates/ParmEd.py:215]: Are these changes necessary if we're killing `ts`?

File: package/MDAnalysis/coordinates/ParmEd.py [hunk 3]
```diff
@@ -174,9 +174,10 @@ def convert(self, obj):
             ag_or_ts = obj.atoms
         except AttributeError:
             if isinstance(obj, base.Timestep):
-                ag_or_ts = obj.copy()
+                raise ValueError("Writing Timesteps to ParmEd "
+                                 "objects is not supported")
             else:
-                raise_from(TypeError("No Timestep found in obj argument"), None)
+                raise_from(TypeError("No atoms found in obj argument"), None)
```
Comment (IAlibay) [inline package/MDAnalysis/coordinates/ParmEd.py:180]: Can you add a test to cover this TypeError please?

**Review by IAlibay — CHANGES_REQUESTED** [review]
Couple of minor things

File: package/MDAnalysis/coordinates/ParmEd.py (hunk 2 above)
Comment (lilyminium) [reply package/MDAnalysis/coordinates/ParmEd.py:215]: I guess I'm anticipating changes in API that don't guarantee resids/masses/whatever, but we can cross that bridge when we come to it. I'll revert it.

File: package/MDAnalysis/coordinates/ParmEd.py [hunk 4]
```diff
@@ -174,9 +174,10 @@ def convert(self, obj):
             ag_or_ts = obj.atoms
         except AttributeError:
             if isinstance(obj, base.Timestep):
-                ag_or_ts = obj.copy()
+                raise ValueError("Writing Timesteps to ParmEd "
+                                 "objects is not supported")
             else:
-                raise_from(TypeError("No Timestep found in obj argument"), None)
+                raise TypeError("No atoms found in obj argument") from None
```
Comment (IAlibay) [inline package/MDAnalysis/coordinates/ParmEd.py:180]: raise_from is needed here unfortunately - py2.7 things

**Review by IAlibay — CHANGES_REQUESTED** [review]
Still gotta live that py2.7 life :(

File: package/MDAnalysis/coordinates/ParmEd.py (hunk 4 above)
Comment (lilyminium) [reply package/MDAnalysis/coordinates/ParmEd.py]: 🐍🐍🐍🐍🐍🐍🐍

**Review by IAlibay — APPROVED** [review]
